File: dataset/ZeroShot - Copy/CLAUDE_claude-sonnet-4-5-20250929/Task159.cpp

#include <iostream>
#include <memory>
#include <stdexcept>

class Task159 {
private:
    static const int BUFFER_SIZE = 100;
    std::unique_ptr<int[]> buffer;
    int size;
    
public:
    Task159() : size(BUFFER_SIZE) {
        buffer = std::make_unique<int[]>(BUFFER_SIZE);
        // Initialize buffer with sample data
        for (int i = 0; i < BUFFER_SIZE; i++) {
            buffer[i] = i * 10;
        }
    }
    
    bool readFromBuffer(int index, int& value) {
        // Secure bounds checking
        if (index < 0 || index >= size) {
            std::cout << "Error: Index out of bounds. Valid range: 0-" 
                      << (size - 1) << std::endl;
            return false;
        }
        value = buffer[index];
        return true;
    }
    
    int getBufferSize() const {
        return size;
    }
};

int main() {
    Task159 task;
    int value;
    
    std::cout << "=== Test Cases ===" << std::endl << std::endl;
    
    // Test case 1: Valid index at start
    std::cout << "Test 1 - Valid index (0):" << std::endl;
    if (task.readFromBuffer(0, value)) {
        std::cout << "Value at index 0: " << value << std::endl;
    }
    std::cout << std::endl;
    
    // Test case 2: Valid index in middle
    std::cout << "Test 2 - Valid index (50):" << std::endl;
    if (task.readFromBuffer(50, value)) {
        std::cout << "Value at index 50: " << value << std::endl;
    }
    std::cout << std::endl;
    
    // Test case 3: Valid index at end
    std::cout << "Test 3 - Valid index (99):" << std::endl;
    if (task.readFromBuffer(99, value)) {
        std::cout << "Value at index 99: " << value << std::endl;
    }
    std::cout << std::endl;
    
    // Test case 4: Invalid negative index
    std::cout << "Test 4 - Invalid negative index (-5):" << std::endl;
    task.readFromBuffer(-5, value);
    std::cout << std::endl;
    
    // Test case 5: Invalid index beyond bounds
    std::cout << "Test 5 - Invalid index beyond bounds (150):" << std::endl;
    task.readFromBuffer(150, value);
    std::cout << std::endl;
    
    return 0;
}

File: dataset/ZeroShot/GEMINI_gemini-2.5-pro/Task159.java
public class Task159 {

    /**
     * Securely reads a byte from the buffer at the given index.
     * This method performs bounds checking to prevent IndexOutOfBoundsException.
     *
     * @param buffer The byte array to read from.
     * @param index  The index of the byte to read.
     * @param outValue A single-element array to store the output value. Using an array
     *                 to simulate an 'out' parameter, allowing the method to signal
     *                 success/failure via its return value.
     * @return true if the read was successful, false otherwise (e.g., out of bounds).
     */
    public static boolean secureRead(byte[] buffer, int index, byte[] outValue) {
        if (buffer == null || outValue == null || outValue.length < 1) {
            // Invalid arguments
            return false;
        }

        // Secure bounds check
        if (index >= 0 && index < buffer.length) {
            outValue[0] = buffer[index];
            return true;
        }

        // Index is out of bounds
        return false;
    }

    public static void main(String[] args) {
        int bufferSize = 10;
        // 1. Dynamically allocate a memory buffer
        byte[] buffer = new byte[bufferSize];

        // Initialize buffer with some data
        for (int i = 0; i < bufferSize; i++) {
            buffer[i] = (byte) (i * 10);
        }

        System.out.println("Buffer created with size: " + bufferSize);
        System.out.println("Buffer content: [0, 10, 20, 30, 40, 50, 60, 70, 80, 90]");
        System.out.println("--- Running Java Test Cases ---");

        // 2. Test cases
        int[] testIndices = {0, 5, 9, -1, 10};

        for (int index : testIndices) {
            byte[] result = new byte[1]; // Holder for the output value
            if (secureRead(buffer, index, result)) {
                System.out.println("Reading at index " + index + ": Success! Value = " + result[0]);
            } else {
                System.out.println("Reading at index " + index + ": Error! Index is out of bounds.");
            }
        }
    }
}